### ISSUE TITLE
Discovering Terraform resources in an Oracle Cloud Infrastructure compartment not working

### ISSUE THREAD

**Issue by eugsim1** [terraform-providers/terraform-provider-oci#922]
hello the discovery fails :
./terraform-provider-oci -command=export -compartment_name=Eugene -output_path=.
2019/11/06 19:53:28 [INFO] terraform-provider-oci 3.50.0
2019/11/06 19:53:29 [INFO] resolved compartment name 'Eugene' to compartment id 'ocid1.compartment.oc1..aaaaaaaa375sfgxnc24b3rmxjju6ttxv264t6ukiyv42txxfxs3zj2difroa'
2019/11/06 19:53:29 Running export command
2019/11/06 19:53:29 [INFO] resource discovery: visiting oci_identity_compartment.export
2019/11/06 19:53:29 [INFO] discovering resources with data source 'oci_identity_availability_domains'
2019/11/06 19:53:29 [INFO] resource discovery: visiting oci_identity_availability_domain.export_WtoZ-EU-FRANKFURT-1-AD-1
2019/11/06 19:53:29 [INFO] discovering resources with data source 'oci_core_boot_volumes'
2019/11/06 19:53:29 [INFO] adding datasource query attribute 'availability_domain' from parent attribute 'name'
2019/11/06 19:53:29 [INFO] resource discovery: visiting oci_identity_availability_domain.export_WtoZ-EU-FRANKFURT-1-AD-2
2019/11/06 19:53:29 [INFO] discovering resources with data source 'oci_core_boot_volumes'
2019/11/06 19:53:29 [INFO] adding datasource query attribute 'availability_domain' from parent attribute 'name'
2019/11/06 19:53:29 [INFO] resource discovery: visiting oci_identity_availability_domain.export_WtoZ-EU-FRANKFURT-1-AD-3
2019/11/06 19:53:29 [INFO] discovering resources with data source 'oci_core_boot_volumes'
2019/11/06 19:53:29 [INFO] adding datasource query attribute 'availability_domain' from parent attribute 'name'
2019/11/06 19:53:29 [INFO] resource discovery: visiting oci_identity_compartment.export
2019/11/06 19:53:29 [INFO] discovering resources with data source 'oci_core_cpes'
2019/11/06 19:53:29 [INFO] discovering resources with data source 'oci_core_cross_connect_groups'
2019/11/06 19:53:29 [INFO] discovering resources with data source 'oci_core_cross_connects'
2019/11/06 19:53:29 [INFO] discovering resources with data source 'oci_core_drg_attachments'
2019/11/06 19:53:29 [INFO] discovering resources with data source 'oci_core_drgs'
2019/11/06 19:53:29 [INFO] discovering resources with data source 'oci_core_images'
2019/11/06 19:53:30 [INFO] discovering resources with data source 'oci_core_instance_configurations'
2019/11/06 19:53:30 [INFO] discovering resources with data source 'oci_core_instance_pools'
2019/11/06 19:53:30 [INFO] discovering resources with data source 'oci_core_instances'
2019/11/06 19:53:32 [WARN] Could not get the boot volume: "Service error:NotAuthorizedOrNotFound. Authorization failed or requested resource not found.. http status code: 404. Opc request id: 21f11da2e692b591ae587ad2ccd4385f/20544E4A5DF863DEF47AAB0C4F28F4D1/4928F94A34E54D82EAD8274C0382D70D"
2019/11/06 19:53:34 [INFO] resource discovery: visiting oci_core_instance.export_Oracle-GoldenGate
2019/11/06 19:53:34 [INFO] discovering resources with data source 'oci_core_volume_backup_policy_assignments'
2019/11/06 19:53:34 [INFO] adding datasource query attribute 'asset_id' from parent attribute 'boot_volume_id'
2019/11/06 19:53:34 [INFO] resource discovery: visiting oci_core_instance.export_database19c
2019/11/06 19:53:34 [INFO] discovering resources with data source 'oci_core_volume_backup_policy_assignments'
2019/11/06 19:53:34 [INFO] adding datasource query attribute 'asset_id' from parent attribute 'boot_volume_id'
2019/11/06 19:53:34 [INFO] resource discovery: visiting oci_core_instance.export_node1
2019/11/06 19:53:34 [INFO] discovering resources with data source 'oci_core_volume_backup_policy_assignments'
2019/11/06 19:53:34 [INFO] adding datasource query attribute 'asset_id' from parent attribute 'boot_volume_id'
2019/11/06 19:53:34 [INFO] resource discovery: visiting oci_core_instance.export_ogg19ma
2019/11/06 19:53:34 [INFO] discovering resources with data source 'oci_core_volume_backup_policy_assignments'
2019/11/06 19:53:34 [INFO] adding datasource query attribute 'asset_id' from parent attribute 'boot_volume_id'
2019/11/06 19:53:34 [INFO] resource discovery: visiting oci_core_instance.export_proxy
2019/11/06 19:53:34 [INFO] discovering resources with data source 'oci_core_volume_backup_policy_assignments'
2019/11/06 19:53:34 [INFO] adding datasource query attribute 'asset_id' from parent attribute 'boot_volume_id'
2019/11/06 19:53:34 [INFO] resource discovery: visiting oci_core_instance.export_pubiptest0
2019/11/06 19:53:34 [INFO] discovering resources with data source 'oci_core_volume_backup_policy_assignments'
2019/11/06 19:53:34 [INFO] adding datasource query attribute 'asset_id' from parent attribute 'boot_volume_id'
2019/11/06 19:53:34 [INFO] resource discovery: visiting oci_core_instance.export_pubiptest1
2019/11/06 19:53:34 [INFO] discovering resources with data source 'oci_core_volume_backup_policy_assignments'
2019/11/06 19:53:34 [INFO] adding datasource query attribute 'asset_id' from parent attribute

**Comment by eugsim1** [comment]
the last line is :
[ERROR] no attribute 'boot_volume_id' found in parent 'oci_core_instance.export_pubiptest1

**Comment by alexng-canuck** [comment]
Hi @eugsim1 , do you happen to have a "STOPPED" instance in your compartment or an instance without an attached boot volume ID in your compartment? There is a known issue in this feature with handling such instances, and a fix should go out in an upcoming release.

**Comment by alexng-canuck** [comment]
Hi @eugsim1 , 

I am not able to repro the exact same error that you had. Would it be possible to run your command with "OCI_GO_SDK_DEBUG=v TF_LOG=DEBUG" environment variables and share the output?

That being said, we now have a fix in the latest version of tool (3.51.0) which will allow the discovery to continue even if it encounters this error. Can you try with the latest version to see if it unblocks your discovery command?

**Comment by eugsim1** [comment]
hi again with the latest provider indeed it worked , but i hit another issue 
my env variable for the region was setup to uk-london-1 with a scrip,
i have check this before to launch the script but it read an existing  config file, and went to frankfurt i had to modify the config and set as default the uk-london-1 
reading the docs unless i m wrong if the TF_VAR_region is setup then the tool should not look to the config file values ?

my terraform script works well with this setting ie i have a env-emea file where i change the region as :
export TF_VAR_region=uk-london-1 and terraforlm created the resources in that region

**Comment by alexng-canuck** [comment]
Hi @eugsim1 , the tool does look for your settings from the config first and then from the TF_VAR variables second. It does not follow the exact same conventions as the Terraform script use case.

However, we are looking at how to make it more consistent with the regular provider use case. For now, assume that the config supersedes other settings.

**Comment by parrneet** [comment]
Closing the issue. Feel free to reopen if you have any more questions regarding this.
